Code_Exercises/helpers.hpp

```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+#include <cstddef> // for size_t
+#include <iostream>
+
+#define SYCLACADEMY_ASSERT(cond)                                               \
+  if (!(cond)) {                                                               \
+    std::cerr << "Failure in " << __FUNCTION__ << std::endl;                   \
```

Comment:
This will print `main` for all tests that just have a `main()` function, which is probably not so useful when multiple tests are run (say, with a script when we are checking that everything works, or in CI) since then we just might get many times `main` when running :)

Perhaps we can look into also printing the filename to avoid this? We'd probably have to use something like gcc `__BASE_FILE__` though (which also works with clang, so hopefully with all SYCL implementations) since `__FILE__` might just print `helpers.hpp` independently of the test.
Or we pass the test name as an additional argument to `SYCLACEDEMY_ASSERT`, but that would clutter the user code a bit more, and look again more like catch2 which we just got rid of.

Reply:
I've changed the impl just to use the `assert` impl (without NDEBUG) as @keryell suggested.